### PR TITLE
Issue 93 - Rake task to create statements when only backdated entries

### PR DESCRIPTION
**Pull request by JBorts**
bitmoji

### CODE REVIEW

**Comment by interlock** [comment]
@JBorts To address #93, where time entries are created after the statement end date, we need to actually make a new rake task that runs on the grace period date instead.
- Check for entries created in period in question, but no associated to a statement
- Create statements for each user who has these entries (a unique user_id should do)
- The CreateStatement interactor will pickup the older entries, which is why we don't need to change the logic of `between` to `before`

**Comment by JBorts** [comment]
I know. But this is an incremental improvement which ensures a statement if generated in the next period and guarantees they will eventually get paid...not something I can say with the current system

File: app/interactors/create_statements_for_period.rb [hunk 1]
```diff
@@ -18,9 +18,9 @@ def set_statement_dates
   end
 
   def fetch_users
-    user_ids = TimeEntry.with_no_statement.between(@from, @to).pluck(:user_id).uniq
+    user_ids = TimeEntry.with_no_statement.before(@to).pluck(:user_id).uniq
```
Comment (interlock) [inline app/interactors/create_statements_for_period.rb]: This line of code is only checking that we have no current statement generated for this period. Not that there COULD be one. That is determined in the `create_statement.rb` interactor.

Comment (JBorts) [inline app/interactors/create_statements_for_period.rb]: This line of code is creating a list of all users that need to have statements generated. This list is then passed into the CreateStatement interactor. 

By switching it from between to before, we will create add the user to the list if there are any outstanding TimeEntries, including those from before this statement period

**Comment by interlock** [comment]
Going to merge this to close #139

**Pull request merged**